frontend/src/components/common/CreateResourceButton.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ClassWithBaseObject, KubeObject } from '../../lib/k8s/cluster';
+import { ActionButton, EditorDialog } from '../common';
+
+export interface CreateResourceButtonProps {
+  resourceClass: ClassWithBaseObject<KubeObject>;
+}
+
+export function CreateResourceButton(props: CreateResourceButtonProps) {
+  const { resourceClass } = props;
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [openDialog, setOpenDialog] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const baseObject = resourceClass.getBaseObject ? resourceClass.getBaseObject() : {};
```

Comment:
This getBaseObject should be available on the base class defined in cluster.ts, so we shouldn't have to check if the function exists. If it gives `null` or something similar by default, we could do:
```typescript
const baseObject = resourceClass.getBaseObject() ?? {};
```

Reply:
Removed the additional check here